File: modules/handlers/auto_delete.py
"""
:project: telegram-onedrive
:author: L-ING
:copyright: (C) 2023 L-ING <hlf01@icloud.com>
:license: MIT, see LICENSE for more details.
"""

from telethon import events
from modules.client import tg_bot
from modules.env import tg_user_name
from modules.utils import check_in_group, check_login


@tg_bot.on(events.NewMessage(pattern="/autoDelete", incoming=True, from_users=tg_user_name))
@check_in_group
@check_login
async def auto_delete_handler(event):
    from modules import env
    if env.delete_flag:
        env.delete_flag = False
        await event.respond("Bot won't auto delete message.")
    else:
        env.delete_flag = True
        await event.respond("Bot will auto delete message.")
    raise events.StopPropagation

File: src/handlers/auto_delete.rs
/*
:project: telegram-onedrive
:author: L-ING
:copyright: (C) 2024 L-ING <hlf01@icloud.com>
:license: MIT, see LICENSE for more details.
*/

use crate::{message::TelegramMessage, state::AppState};
use anyhow::{Context, Result};
use proc_macros::{check_in_group, check_senders};
use std::sync::atomic::Ordering;

pub const PATTERN: &str = "/autoDelete";

#[check_senders]
#[check_in_group]
pub async fn handler(message: TelegramMessage, state: AppState) -> Result<()> {
    let should_auto_delete = state.should_auto_delete.load(Ordering::Acquire);

    state
        .should_auto_delete
        .store(!should_auto_delete, Ordering::Release);

    if should_auto_delete {
        let response = "Bot won't auto delete message.";
        message.respond(response).await.context(response)?;
    } else {
        let response = "Bot will auto delete message.";
        message.respond(response).await.context(response)?;
    }

    Ok(())
}
